modules/createHashHistory.js

```diff
@@ -33,7 +37,7 @@ var DefaultQueryKey = '_k';
 function createHashHistory(options={}) {
   var { queryKey } = options;
 
-  if (queryKey && typeof queryKey !== 'string')
+  if (typeof queryKey !== 'string')
```

Comment:
It can be done with the original line but then we need to add check to finishTransition to not save state if queryKey is undefined, because it will fail tests.